cmd/frontend/graphqlbackend/repository_comparison.go

```diff
@@ -373,6 +411,63 @@ func (r *DiffHunk) Section() *string {
 
 func (r *DiffHunk) Body() string { return string(r.hunk.Body) }
 
+type richHunk struct {
+	line string
+	kind string
+}
+
+func (r *richHunk) Line() string {
+	return r.line
+}
+
+func (r *richHunk) Kind() string {
+	return r.kind
+}
+
+func (r *DiffHunk) RichBody() []*richHunk {
+	hunkLines := strings.Split(string(r.hunk.Body), "\n")
+	richHunks := make([]*richHunk, len(hunkLines))
+	var baseLine = r.hunk.OrigStartLine - 1
+	var headLine = r.hunk.NewStartLine - 1
+	for i, hunkLine := range hunkLines {
+		var kind string
+		if len(hunkLine) == 0 {
+			kind = "unchanged"
+			baseLine++
+			headLine++
+		} else if hunkLine[0] == '+' {
+			kind = "addition"
+			headLine++
+		} else if hunkLine[0] == '-' {
+			kind = "deletion"
+			baseLine++
+		} else {
+			kind = "unchanged"
+			baseLine++
+			headLine++
+		}
+
+		var line string
+		switch kind {
+		case "unchanged":
+			hl := (*r.highlightedBase)[baseLine]
+			line = hl[:5] + " " + hl[5:]
+		case "deletion":
+			hl := (*r.highlightedBase)[baseLine]
+			line = hl[:5] + "-" + hl[5:]
+		case "addition":
+			hl := (*r.highlightedHead)[headLine]
+			line = hl[:5] + "+" + hl[5:]
+		}
```

Comment:
Where does the `5` come from? What if `hl` is shorter?

Also, I think this could be simplified:

```suggestion
		var hl, prefix string
		switch kind {
		case "unchanged":
			hl = r.highlightedBase[baseLine]
			prefix = " "
		case "deletion":
			hl = r.highlightedBase[baseLine]
			prefix = "-"
		case "addition":
			hl = r.highlightedHead[headLine]
			prefix = "+"
		}
		line := hl[:5] + prefix + hl[5:]
```

Reply:
It's always wrapped in a `<div>`, that's where the 5 came from. It's not ideal but worked..